enhancements/platform-validation.md

```diff
@@ -0,0 +1,195 @@
+
+Since the Kind is Kubernetes cluster running in Docker container, it may be difficult for us to directly check the Kepler exposed metrics on port 9102.
+
+The easier way is to deploy Prometheus service and Kepler related Service Monitor in the cluster. Expose Prometheus service as `NodePort` type.
+Codes/scripts could then retrieve what we needed data from Prometheus using either REST API calling or PromQL queries.
+
+A typical but not only way is could be found in `Test Plan` section below.
+
+### Risks and Mitigations
+
+
+### Drawbacks
+
+
+## Design Details
+
+### Test Plan
+
+The initial test case is only for CPU architecture recognition.
+
+When we check the `cpu_architecture` label exported by Kepler, we do below things:
+
+1. Deploy Kepler with Prometheus.
+
+```
+make build-manifest OPTS="PROMETHEUS_DEPLOY"
+```
+
+2. Change the Prometheus service as `NodePort` type and get the accessible port.
+
+```
+kubectl patch svc prometheus-k8s -n monitoring --type='json' -p '[{"op":"replace","path":"/spec/type","value":"NodePort"}]'
+```
+```
+$kubectl get svc prometheus-k8s -n monitoring
+NAME             TYPE       CLUSTER-IP    EXTERNAL-IP   PORT(S)                         AGE
+prometheus-k8s   NodePort   10.96.205.9   <none>        9090:32517/TCP,8080:30331/TCP   5d9h
+```
+
+Tools such as `awk` could retrieve the accessible port:
+```
+$kubectl get svc prometheus-k8s -n monitoring | grep 9090 | awk '{print $5}' | awk -F ',' '{print $1}' | awk -F '/' '{print substr($1,6)}'
+32517
+```
+
+3. Get the test node's IP.
+
+```
+$ kubectl get nodes -o wide
+NAME                 STATUS   ROLES           AGE     VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
+kind-control-plane   Ready    control-plane   5d10h   v1.25.3   172.19.0.2    <none>        Ubuntu 22.04.1 LTS   5.19.0-41-generic   containerd://1.6.9
+```
+
+```
+$ kubectl get nodes -o wide |grep control-plane | awk '{print $6}'
+172.19.0.2
+```
+
+4. Retrieve the metrics with `cpu_architecture` label in Prometheus.
+```
+$ curl --noproxy "*" -s http://172.19.0.2:32517/api/v1/label/cpu_architecture/values | jq -r ".data[]" | sort
+Sapphire Rapids
+```
+
+5. Check the actual `cpu_architure` info on the node:
+
+Per current Kepler code,
+
+(1) For x86 platforms, we use `cpuid` command to fetch the cpu arch info:
+```
+$ cpuid -1 |grep uarch
+   (uarch synth) = Intel Sapphire Rapids {Sunny Cove}, 10nm+
+```
+
+(2) For ARM platforms, we use `archspec` Python tool, follow the usage guide in this [repo](https://github.com/archspec/archspec)
+
+(3) For s390x platforms, we use `lscpu` command to check the machine type:
+```
+$ lscpu |grep "Machine type"
+Machine type:        3906
+```
+
+
+The further test cases include not but limited to:
+
+1. Typical workloads power consumption measurement (carbon footprint) accuracy check on specific platform.
+2. Carbon footprint accuracy check in scaling scenarios (VPA/HPA) on specific platform.
```

Comment:
I want to query here.
as cpu_architure is a specific metric, and the further case seems cover k8s feature as VPA? I see mismatch or different level of things.

Reply:
`cpu_architecture` is not metric, but one of the labels of some metrics, we chose it as the first test case because it can somehow stands for the platform.
For VPA, may introduce other metrics or other labels check then.